internal/filesystem.go

```diff
@@ -107,6 +108,43 @@ func (rdl *runDirListener) Addr() net.Addr {
 // ManualListener returns manually created listener with provided permissions
 func ManualListener(socket string, perm fs.FileMode) func() (net.Listener, error) {
 	return func() (net.Listener, error) {
+		pidfile := DaemonPid
+		// check and read pid file
+		if FileExists(pidfile) {
+			out, err := FileRead(pidfile)
+			if err != nil {
+				// pid file exists, but is not readable,
+				// some garbage from previous run/failure? maybe we can cleanup and proceed?
+				return nil, fmt.Errorf("daemon pid file is not readable: %w", err)
+			}
+			pidFromFile, err := strconv.Atoi(strings.TrimSpace(string(out)))
+			if err != nil {
+				// pid value is not valid integer, some garbage from previous run/failure?
+				log.Println(WarningPrefix, fmt.Errorf("daemon pid file does not contain valid integer value: %w", err))
+			} else {
+				// try to find existing process in process list by pid
+				processes, _ := process.Processes()
```

Comment:
There is no need to list all processes. Simply check for `/proc/<pid>` directory

Reply:
Yes, right, done